src/plugin/cesium/cesium.js

```diff
@@ -403,37 +417,136 @@ plugin.cesium.updateCesiumLayerProperties = function(olLayer, csLayer) {
  * Create a Cesium terrain provider instance.
  *
  * @param {Cesium.CesiumTerrainProviderOptions} options The Cesium terrain options.
- * @return {!Cesium.CesiumTerrainProvider}
+ * @return {!goog.Promise<!Cesium.CesiumTerrainProvider>}
  */
 plugin.cesium.createCesiumTerrain = function(options) {
-  return new Cesium.CesiumTerrainProvider(options);
+  return goog.Promise.resolve(new Cesium.CesiumTerrainProvider(options));
 };
 
 
 /**
  * Create a Cesium World Terrain instance.
  *
  * @param {Cesium.WorldTerrainOptions} options The Cesium World Terrain options.
- * @return {!Cesium.CesiumTerrainProvider}
+ * @return {!goog.Promise<!Cesium.CesiumTerrainProvider>}
  */
 plugin.cesium.createWorldTerrain = function(options) {
   var assetId = options.assetId != null ? options.assetId : 1;
-  return plugin.cesium.createCesiumTerrain({
+  var accessToken = options.accessToken || /** @type {string|undefined} */ (
+    os.settings.get(plugin.cesium.SettingsKey.ACCESS_TOKEN));
+
+  if (!accessToken) {
+    return plugin.cesium.promptForAccessToken().then((accessToken) => {
+      return plugin.cesium.createWorldTerrain_(assetId, accessToken);
+    });
+  } else {
+    return goog.Promise.resolve(plugin.cesium.createWorldTerrain_(assetId, accessToken));
+  }
+};
+
+
+/**
+ * Create a Cesium World Terrain instance.
+ *
+ * @param {number} assetId The Cesium World Terrain asset id.
+ * @param {string} accessToken The Cesium World Terrain access token.
+ * @return {!Cesium.CesiumTerrainProvider}
+ *
+ * @private
+ */
+plugin.cesium.createWorldTerrain_ = function(assetId, accessToken) {
+  return new Cesium.CesiumTerrainProvider({
     url: Cesium.IonResource.fromAssetId(assetId, {
-      accessToken: options.accessToken
+      accessToken
     })
   });
 };
 
 
+/**
+ * Prompt the user to activate Cesium World Terrain.
+ * @param {string} prompt The message to display.
+ */
+plugin.cesium.promptForWorldTerrain = function(prompt) {
+  const showPrompt = os.settings.get(plugin.cesium.SettingsKey.SHOW_TERRAIN_PROMPT, true);
+  if (showPrompt && !plugin.cesium.isWorldTerrainActive() && plugin.cesium.hasWorldTerrain()) {
+    os.ui.window.ConfirmUI.launchConfirm(/** @type {!osx.window.ConfirmTextOptions} */ ({
+      confirm: plugin.cesium.enableWorldTerrain,
+      cancel: () => {
+        // Stop asking if the user says no
+        os.settings.set(plugin.cesium.SettingsKey.SHOW_TERRAIN_PROMPT, false);
```

Comment:
Will this cause the prompt to never pop up again?

Reply:
Unless they reset settings, yes. I wasn't sure how annoying I wanted this to be. Hopefully they click Yes because OSM Buildings in particular are awful if the terrain doesn't line up, but if they click No should we continue telling them whenever the layer is activated (including app refresh)? I could go either way.

Reply:
I decided to remove this. If we want something like this, we should either implement a "remember this decision" or "don't ask me again" type checkbox on `confirmtext`.